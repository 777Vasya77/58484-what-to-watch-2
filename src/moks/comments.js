export const COMMENTS = [
  {
    'id': 1,
    'user': {
      'id': 18,
      'name': `Sophie`
    },
    'rating': 2,
    'comment': `Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.`,
    'date': `2019-10-18T19:05:06.212Z`
  },
  {
    'id': 2,
    'user': {
      'id': 16,
      'name': `Mollie`
    },
    'rating': 2,
    'comment': `This movie really touched my heart, it really is the best movie of the year and everyone should see this masterpiece.`,
    'date': `2019-10-29T19:05:06.212Z`
  },
  {
    'id': 3,
    'user': {
      'id': 14,
      'name': `Corey`
    },
    'rating': 5,
    'comment': `A movie that will take you to another world full of emotions.`,
    'date': `2019-10-02T19:05:06.212Z`
  },
  {
    'id': 4,
    'user': {
      'id': 18,
      'name': `Sophie`
    },
    'rating': 4,
    'comment': `This movie is perfect in all its categories: credits, sound track, production, casting, writing, photography, editing, acting, and direction.\nI was amazed with the freedom of the use of the camera. This movie will change the way movies are made. Slow-mo, stills, black and white, and color were all used to brilliant effect.`,
    'date': `2019-10-27T19:05:06.212Z`
  },
  {
    'id': 5,
    'user': {
      'id': 18,
      'name': `Sophie`
    },
    'rating': 5,
    'comment': `I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.`,
    'date': `2019-10-20T19:05:06.212Z`
  },
  {
    'id': 6,
    'user': {
      'id': 14,
      'name': `Corey`
    },
    'rating': 5,
    'comment': `I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.`,
    'date': `2019-10-20T19:05:06.212Z`
  },
  {
    'id': 7,
    'user': {
      'id': 13,
      'name': `Zak`
    },
    'rating': 3,
    'comment': `Unfortunately we don't have a reliable way to tell the true ratings of a movie.`,
    'date': `2019-10-26T19:05:06.212Z`
  },

];
