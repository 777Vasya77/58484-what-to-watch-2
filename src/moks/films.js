export const FILMS = [
  {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    image: `fantastic-beasts-the-crimes-of-grindelwald.jpg`
  },
  {
    title: `Aviator`,
    image: `aviator.jpg`
  },
  {
    title: `Bohemian Rhapsody`,
    image: `bohemian-rhapsody.jpg`
  },
  {
    title: `Macbeth`,
    image: `macbeth.jpg`
  },
  {
    title: `Mindhunter`,
    image: `mindhunter.jpg`
  },
  {
    title: `War of the Worlds`,
    image: `war-of-the-worlds.jpg`
  },
  {
    title: `Johnny English`,
    image: `johnny-english.jpg`
  },
  {
    title: `We Need to Talk about Kevin`,
    image: `we-need-to-talk-about-kevin.jpg`
  },
];
