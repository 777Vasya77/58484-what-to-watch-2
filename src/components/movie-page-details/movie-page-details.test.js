import React from 'react';
import renderer from 'react-test-renderer';
import MoviePageDetails from '~/components/movie-page-details/movie-page-details';
import Film from '~/models/film';
import {FILMS} from '~/moks/test-moks';

const film = new Film(FILMS[0]);
const similarFilms = Film.parseFilms(FILMS).slice(4);

describe(`MoviePageReviews component tests`, () => {
  it(`Component render correctly`, () => {
    const tree = renderer
        .create(
            <MoviePageDetails
              film={film}
              similarFilms={similarFilms}
            />,
            {createNodeMock: () => ({})}
        ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});


