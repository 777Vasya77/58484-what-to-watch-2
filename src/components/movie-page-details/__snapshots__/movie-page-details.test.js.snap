// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePageReviews component tests Component render correctly 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        Justin Kurzel
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
      >
        Michael Fassbender,
Marion Cotillard,
Jack Madigan
      </span>
    </p>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
      >
        1h 53m
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        Drama
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        2015
      </span>
    </p>
  </div>
</div>
`;
